Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/PosPart.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.NonUnital
+import Mathlib.Topology.ContinuousMap.StarOrdered
+
+/-! # The positive (and negative) parts of a selfadjoint element in a C⋆-algebra
+
+This file defines the positive and negative parts of a selfadjoint element in a C⋆-algebra via
+the continuous functional calculus and develops the basic API.
+
+## TODO
+
++ Prove that the positive and negative parts are unique.
+
+-/
+
+variable {A : Type*} [NonUnitalRing A] [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
+variable [StarRing A] [TopologicalSpace A]
+variable [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
+
+namespace CStarAlgebra
+
+noncomputable instance : PosPart A where
+  posPart := cfcₙ (·⁺ : ℝ → ℝ)
+
+noncomputable instance : NegPart A where
+  negPart := cfcₙ (·⁻ : ℝ → ℝ)
+
+end CStarAlgebra
+
+attribute [fun_prop] continuous_posPart continuous_negPart
+
+namespace CFC
+
+lemma posPart_def (a : A) : a⁺ = cfcₙ (·⁺ : ℝ → ℝ) a := rfl
+
+lemma negPart_def (a : A) : a⁻ = cfcₙ (·⁻ : ℝ → ℝ) a := rfl
+
+@[simp]
+lemma posPart_mul_negPart (a : A) : a⁺ * a⁻ = 0 := by
+  rw [posPart_def, negPart_def]
+  by_cases ha : IsSelfAdjoint a
+  · rw [← cfcₙ_mul _ _, ← cfcₙ_zero ℝ a]
+    refine cfcₙ_congr (fun x _ ↦ ?_)
+    simp only [_root_.posPart_def, _root_.negPart_def]
+    simpa using le_total x 0
+  · simp [cfcₙ_apply_of_not_predicate a ha]
+
+@[simp]
+lemma negPart_mul_posPart (a : A) : a⁻ * a⁺ = 0 := by
+  rw [posPart_def, negPart_def]
+  by_cases ha : IsSelfAdjoint a
+  · rw [← cfcₙ_mul _ _, ← cfcₙ_zero ℝ a]
+    refine cfcₙ_congr (fun x _ ↦ ?_)
+    simp only [_root_.posPart_def, _root_.negPart_def]
+    simpa using le_total 0 x
+  · simp [cfcₙ_apply_of_not_predicate a ha]
+
+lemma posPart_sub_negPart (a : A) (ha : IsSelfAdjoint a := by cfc_tac) : a⁺ - a⁻ = a := by
+  rw [posPart_def, negPart_def]
+  rw [← cfcₙ_sub _ _]
+  conv_rhs => rw [← cfcₙ_id ℝ a]
+  congr! 2 with
+  exact _root_.posPart_sub_negPart _
+
+section Unique
+
+variable [UniqueNonUnitalContinuousFunctionalCalculus ℝ A]
+
+@[simp]
+lemma posPart_neg (a : A) : (-a)⁺ = a⁻ := by
+  by_cases ha : IsSelfAdjoint a
+  · rw [posPart_def, negPart_def, ← cfcₙ_comp_neg _ _]
+    congr! 2
+  · have ha' : ¬ IsSelfAdjoint (-a) := fun h ↦ ha (by simpa using h.neg)
+    rw [posPart_def, negPart_def, cfcₙ_apply_of_not_predicate a ha,
+      cfcₙ_apply_of_not_predicate _ ha']
+
+@[simp]
+lemma negPart_neg (a : A) : (-a)⁻ = a⁺ := by
+  rw [← eq_comm, ← sub_eq_zero, ← posPart_neg, neg_neg, sub_self]
+
+end Unique
+
+variable [PartialOrder A] [StarOrderedRing A]
+
+lemma posPart_nonneg (a : A) :
+    0 ≤ a⁺ :=
```

Comment:
```suggestion
@[simp] lemma posPart_nonneg (a : A) :
    0 ≤ a⁺ :=
```
(and the same for negPart). Or perhaps `@[aesop norm apply (rule_sets := [CStarAlgebra])]` to avoid triggering a search for a CFC where it's not relevant.